### PR TITLE
Iav arm musl dessalines

### PR DESCRIPTION
**Pull request by dessalines**
This renames iav's `Dockerfile.m` and moves it to `docker/prod/Dockerfile`. 

I'll update the deploy script to make sure it only does the full deploy when its not a semver patch (3rd semver) deploy.

### CODE REVIEW

**Comment by dessalines** [comment]
Okay @Nutomic this is ready to go, it just needs a quick once over. I tested the deploy.sh script and it works.

**Comment by Nutomic** [comment]
Please delete the old Dockerfiles in `docker/prod/docker_old/`, they are in the git history if we ever need them. Also you should probably remove the commented lines in `docker/prod/Dockerfile`, they are just confusing. And you probably dont want to change the version in this PR.

**Comment by Nutomic** [comment]
The x86 build works fine for me as I mentioned, but I didnt manage to setup the cross-compile (like I mentioned in the other issue). We probably need some docs for that.

File: docker/prod/Dockerfile [hunk 1]
```diff
@@ -1,61 +1,63 @@
...
+   && echo 'fn main() { println!("Dummy") }' > ./src/bin/main.rs
+RUN cargo build --release
+RUN rm -f ./target/$CARGO_BUILD_TARGET/$RUSTRELEASEDIR/deps/lemmy_server*
+COPY --chown=rust:rust server/src ./src/
+COPY --chown=rust:rust server/migrations ./migrations/
 
-# Build for release
-RUN RUSTFLAGS="-Clink-arg=-fuse-ld=lld" cargo build --frozen --release
+# build for release
+# workaround for https://github.com/rust-lang/rust/issues/62896
```
Comment (Nutomic) [inline docker/prod/Dockerfile:23]: Not sure what this comment is about, why would we not build for release here?

**Pull request merged**